lib/vdsm/storage/lvm.py

```diff
@@ -910,29 +916,39 @@ def getPvs(self, vgName):
         return pvs
 
     def getVg(self, vgName):
-        # Get specific VG
+        """
+        Get specific VG.
+        Raise a VolumeGroupDoesNotExist for LVM command errors or missing VG.
+        """
         vg = self._vgs.get(vgName)
         if not vg or vg.is_stale():
             self.stats.miss()
             vgs = self._reload_single_vg(vgName)
+            if vgName not in vgs:
+                raise se.VolumeGroupDoesNotExist(vg_name=vgName)
             vg = vgs.get(vgName)
         else:
             self.stats.hit()
         return vg
 
     def getVgs(self, vgNames):
-        """Reloads all the VGs of the set.
+        """
+        Reload all the VGs of the set.
 
         Can block for suspended devices.
-        Fills the cache but not uses it.
-        Only returns found VGs.
+        Fill the cache but do not use it.
+        Return only found VGs.
         """
         self.stats.miss()
         return [vg for vgName, vg in self._reloadvgs(vgNames).items()
                 if vgName in vgNames]
 
     def getAllVgs(self):
-        # Get everything we have
+        """
+        Get everything we have.
+        Reload all VGs. Even on cache hits, reload stale VGs.
+        Return only found VGs.
+        """
```

Comment:
Since we don't need to modify this code now, better not touch the docstring now. We can do this later in a cleanup PR.